analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -138,9 +138,11 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
                     nonEmptyAccess.Add(operation);
                 }
             }
-            if (AddMethods.Contains(invocation.TargetMethod.Name) && invocation.Instance.TrackedSymbol() is { } symbol)
+            if (AddMethods.Contains(invocation.TargetMethod.Name))
             {
-                return context.State.SetSymbolConstraint(symbol, CollectionConstraint.NotEmpty);
+                return invocation.Instance.TrackedSymbol() is { } symbol
+                    ? context.State.SetSymbolConstraint(symbol, CollectionConstraint.NotEmpty)
+                    : context.State.SetOperationConstraint(invocation.Instance, CollectionConstraint.NotEmpty);
```

Comment:
`SetOperationConstraint` should happen in any case.
```suggestion
                var state = context.State.SetOperationConstraint(invocation.Instance, CollectionConstraint.NotEmpty);
                return invocation.Instance.TrackedSymbol() is { } symbol
                    ? state.SetSymbolConstraint(symbol, CollectionConstraint.NotEmpty)
                    : state;
```

Reply:
I found a regular if to be significantly easier to understand.